# Description: checks if ppc_md.feature_call is present before using it.
# Patch author: David Woodhouse <dwmw2@infradead.org>
# Upstream status: submitted.

We shouldn't be assuming that ppc_md.feature_call will be present.

Signed-off-by: David Woodhouse <dwmw2@infradead.org>

--- linux-2.6.12/sound/ppc/pmac.c.orig	2005-08-17 11:19:46.000000000 +0100
+++ linux-2.6.12/sound/ppc/pmac.c	2005-08-17 11:20:05.000000000 +0100
@@ -765,7 +765,8 @@ snd_pmac_ctrl_intr(int irq, void *devid,
  */
 static void snd_pmac_sound_feature(pmac_t *chip, int enable)
 {
-	ppc_md.feature_call(PMAC_FTR_SOUND_CHIP_ENABLE, chip->node, 0, enable);
+	if (ppc_md.feature_call)
+		ppc_md.feature_call(PMAC_FTR_SOUND_CHIP_ENABLE, chip->node, 0, enable);
 }
 
 /*

-- 
dwmw2

---------------------------------------------------------------------------------------
Wanadoo vous informe que cet  e-mail a ete controle par l'anti-virus mail. 
Aucun virus connu a ce jour par nos services n'a ete detecte.



